src/query/queryWithMultiIndex.ts

```diff
@@ -0,0 +1,127 @@
+import { DocumentClient, Key, QueryOutput } from 'aws-sdk/clients/dynamodb';
+import { decrypt, encrypt } from '../utils';
+import { AnyObject, Filter, TableConfig } from '../types';
+import { buildQueryTableParams } from './queryBuilder';
+import { keyBy } from 'lodash';
+
+const DEFAULT_LIMIT = 99999;
+
+export type Queries = {
+  filter: Filter;
+  indexName?: string;
+};
+
+export interface Cursor {
+  [indexName: string]: Key;
+}
+
+export interface Outputs extends QueryOutput {
+  indexName?: string;
+}
+
+/**
+ * Cursor based query which returns list of matching items and the last cursor position
+ *
+ * @param {Queries} queries array of query filter with index
+ * @param { QueryWithCursorOptions } options cursor, page size options
+ * @param {number} limit number of items
+ * @param {string} prevCursor starting cursor position
+ * @returns {Array<T>, string} list of matching items, last cursor position
+ */
+export async function queryWithMultiIndex<T extends AnyObject>(
+  dbClient: DocumentClient,
+  table: TableConfig,
+  queries: Queries[],
+  limit?: number,
+  prevCursor?: string,
+): Promise<{ items: Array<T>; cursor?: string; scannedCount: number }> {
+  if (!table.cursorSecret) {
+    throw new Error(
+      'Expected `cursorSecret` which is used to encrypt the `LastEvaluatedKey`',
+    );
+  }
+  let queryParams = [];
+  const cursor = prevCursor
+    ? decrypt<Cursor>(prevCursor, table.cursorSecret)
+    : {};
+  // process each query
+  for (const query of queries) {
+    const { partitionKeyName, sortKeyName } = table.indexes[
```

Comment:
Can we say partitionKey instead of partitionKeyName? 
same for sortKeyName

Reply:
it's the name of partionKey like gs1pk, or something else.
changing the name here will break our existing system.
we have to change everywhere in tillx as well as in this repository.